src/algorithm/retrieveWebidTrustedOidcIssuers.ts

```diff
@@ -1,48 +1,61 @@
-import { DataFactory } from "n3";
-import rdfDereferencer from "rdf-dereference";
-import type { Quad, Stream } from "rdf-js";
+// eslint-disable-next-line no-shadow
+import { URL } from "url";
+import type { Quad } from "@rdfjs/types";
+import { DataFactory, Parser, Store } from "n3";
+// eslint-disable-next-line no-shadow
+import fetch from "node-fetch";
 import { WebidDereferencingError } from "../error/WebidDereferencingError";
+import { WebidIriError } from "../error/WebidIriError";
 import { WebidParsingError } from "../error/WebidParsingError";
 import type { RetrieveOidcIssuersFunction } from "../type";
 
-const defaultGraph = DataFactory.defaultGraph();
-const oidcIssuer = DataFactory.namedNode(
-  "http://www.w3.org/ns/solid/terms#oidcIssuer"
-);
-
-async function dereferenceWebid(webid: string): Promise<Stream<Quad>> {
+async function dereferenceWebid(webid: string): Promise<string> {
   try {
-    return (await rdfDereferencer.dereference(webid)).quads;
+    const response = await fetch(webid, {
+      headers: {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        Accept: "text/turtle",
```

Comment:
Is a WebID required to be hosted somewhere that supports returning it as turtle? I know a Solid server is required to support turtle for RDF resources, but can a WebID also be hosted as a static file on a non-Solid server? Because this would break if a WebID is only stored as JSON-LD for example. Genuine question since I don't know what the exact spec requirements are for a WebID.

Reply:
It is not. Have encountered this issue before, will flag it and link.

Reply:
https://github.com/solid/specification/issues/376

Reply:
The turtle serialisation is required because it is one of the [three requirements in the WebID spec](https://notes.cognithive.com/webid#requirements).

Thanks for reporting the issue Ruben! I repplied to the thread there https://github.com/solid/specification/issues/376#issuecomment-1021414071.

Reply:
>  but can a WebID also be hosted as a static file on a non-Solid server? Because this would break if a WebID is only stored as JSON-LD for example.

Specifically, if it was hosted as a static document that isn't content-negotiable as turtle from the WebID URL, it would not be a WebID (or a WebID profile document) according to the WebID spec.